src/client/components/ProjectDetailsModal.tsx

```diff
@@ -0,0 +1,324 @@
+/** @jsx jsx */
+import React, { useState } from "react";
+import AriaModal from "react-aria-modal";
+import { connect } from "react-redux";
+import { InputField } from "./Field";
+import {
+  Box,
+  Button,
+  Flex,
+  Heading,
+  jsx,
+  ThemeUIStyleObject,
+  Label,
+  Checkbox,
+  Divider
+} from "theme-ui";
+
+import { IProject } from "../../shared/entities";
+import { State } from "../reducers";
+import store from "../store";
+import { toggleProjectDetailsModal, updateProjectDetailsSuccess } from "../actions/projectData";
+import { WriteResource } from "../resource";
+import FormError from "./FormError";
+import { DistrictsGeoJSON } from "../types";
+import MultiMemberForm from "./MultiMemberForm";
+import { patchProject } from "../api";
+
+const style: ThemeUIStyleObject = {
+  footer: {
+    marginTop: 5
+  },
+  header: {
+    mb: 5
+  },
+  heading: {
+    fontFamily: "heading",
+    fontWeight: "bold",
+    fontSize: 4
+  },
+  helpText: {
+    fontSize: "13px",
+    textTransform: "none",
+    fontWeight: "normal"
+  },
+  label: {
+    fontSize: "16px",
+    color: "heading",
+    fontWeight: "bold",
+    textTransform: "none"
+  },
+  modal: {
+    bg: "muted",
+    p: 5,
+    width: "small",
+    maxWidth: "90vw",
+    overflow: "visible"
+  },
+  customInputContainer: {
+    width: "100%",
+    mb: 5
+  },
+  multiMemberContainer: {
+    overflow: "hidden",
+    maxHeight: "calc(100vh - 42rem)",
+    width: "100%",
+    "> div": {
+      overflow: "auto",
+      width: "100%",
+      flex: "0 0 auto",
+      maxHeight: "inherit"
+    },
+    thead: {
+      position: "sticky",
+      top: 0
+    }
+  }
+};
+
+const validate = (form: ConfigurableForm): ProjectDetailsForm => {
+  const name = form.name;
+  const populationDeviation = form.populationDeviation;
+  const numberOfMembers = form.numberOfMembers;
+  const isMultiMember = form.isMultiMember;
+
+  return name && populationDeviation !== null && numberOfMembers !== null
+    ? {
+        name,
+        populationDeviation,
+        numberOfMembers,
+        isMultiMember,
+        valid: true
+      }
+    : {
+        name,
+        populationDeviation,
+        numberOfMembers,
+        isMultiMember,
+        valid: false
+      };
```

Comment:
Not sure it's actually better, but this could be rewritten to put the condition inline in the object, since the value of `valid` is the only thing actually affected by the condition.
```suggestion
  return {
        name,
        populationDeviation,
        numberOfMembers,
        isMultiMember,
        valid: name && populationDeviation !== null && numberOfMembers !== null
  };
```

Reply:
If I try this out, I get type errors.

I think our being verbose here helps the TS compiler see that we're returning either a `ValidForm` or an `InvalidForm`, but it may be getting confused for some other reason.

Reply:
Doh, yeah, I didn't try it out, just read it, squinted at it to see what was different, then wrote this down.  It does seem like in a perfect world the type checker would be able to recognize that this still comes out to `ValidForm | InvalidForm`, but yeah, not today.